yb-voyager/src/srcdb/postgres.go

```diff
@@ -104,6 +112,502 @@ func (pg *PostgreSQL) Disconnect() {
+	rows, err := pg.db.Query(checkTableOwnerPermissionQuery)
+	if err != nil {
+		return nil, fmt.Errorf("error in querying(%q) source database for checking table owner permission: %w", checkTableOwnerPermissionQuery, err)
+	}
+	defer func() {
+		closeErr := rows.Close()
+		if closeErr != nil {
+			log.Warnf("close rows for query %q: %v", checkTableOwnerPermissionQuery, closeErr)
+		}
+	}()
+
+	var missingTables []string
+	var tableSchemaName, tableName, ownerName string
+	var hasOwnership bool
+
+	for rows.Next() {
+		err = rows.Scan(&tableSchemaName, &tableName, &ownerName, &hasOwnership)
+		if err != nil {
+			return nil, fmt.Errorf("error in scanning query rows for table names: %w", err)
+		}
+		if !hasOwnership {
+			missingTables = append(missingTables, fmt.Sprintf("%s.%s", tableSchemaName, tableName))
+		}
+	}
+
+	// Check for errors during row iteration
+	if err = rows.Err(); err != nil {
+		return nil, fmt.Errorf("error iterating over query rows: %w", err)
+	}
+
+	return missingTables, nil
+}
+
+func (pg *PostgreSQL) checkCreatePermissionOnDB() (bool, error) {
+	query := `SELECT
+	EXISTS (
+		SELECT 1
+		FROM pg_database
+		WHERE datname = current_database()
+		  AND has_database_privilege($1, datname, 'CREATE')
+	) AS has_create_permission;`
+	var hasCreatePermission bool
+	err := pg.db.QueryRow(query, pg.source.User).Scan(&hasCreatePermission)
+	if err != nil {
+		return false, fmt.Errorf("error in checking create permission: %w", err)
+	}
+	return hasCreatePermission, nil
+}
+
+func (pg *PostgreSQL) checkReplicationPermission() (bool, error) {
+	query := `WITH instance_check AS (
+		SELECT
+			CASE
+				WHEN EXISTS (SELECT 1 FROM pg_roles WHERE rolname = 'rds_superuser')
+				THEN 'rds'
+				ELSE 'standalone'
+			END AS db_instance_type
+	)
+	SELECT
+		CASE
+			WHEN db_instance_type = 'rds' THEN
+				EXISTS (
+					SELECT 1
+					FROM pg_roles
+					WHERE rolname = $1
+					  AND pg_has_role($1, 'rds_replication', 'USAGE')
+				)
+			ELSE
+				EXISTS (
+					SELECT 1
+					FROM pg_roles
+					WHERE rolname = $1
+					  AND rolreplication
+				)
+		END AS has_permission
+	FROM instance_check;`
+
+	var hasPermission bool
+	err := pg.db.QueryRow(query, pg.source.User).Scan(&hasPermission)
+	if err != nil {
+		return false, fmt.Errorf("error in checking replication permission: %w", err)
+	}
+	return hasPermission, nil
+}
+
+func (pg *PostgreSQL) checkWhichTablesDontHaveReplicaIdentityFull() ([]string, error) {
+	trimmedSchemaList := pg.getTrimmedSchemaList()
+	querySchemaList := "'" + strings.Join(trimmedSchemaList, "','") + "'"
+	checkTableReplicaIdentityQuery := fmt.Sprintf(`SELECT
+	n.nspname AS schema_name,
+	c.relname AS table_name,
+	c.relreplident AS replica_identity,
+	CASE WHEN c.relreplident <> 'f' THEN 'Missing FULL' ELSE 'Correct' END AS status
+FROM pg_class c
+JOIN pg_namespace n ON c.relnamespace = n.oid
+WHERE quote_ident(n.nspname) IN (%s)
+	AND c.relkind = 'r';`, querySchemaList)
```

Comment:
hmm, now i am thinking that do we need to check on parent if child has permissions.
Because if i remember correctly, we only deal with partitions(leaf nodes) nothing else, during export data.
cc @priyanshi-yb